pkg/controller/encryption/ipsec/ipsec_controller.go

```diff
@@ -0,0 +1,554 @@
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/util/workqueue"
+	"kmesh.net/kmesh/pkg/bpf/restart"
+	"kmesh.net/kmesh/pkg/constants"
+	kmesh_netns "kmesh.net/kmesh/pkg/controller/netns"
+	"kmesh.net/kmesh/pkg/kube"
+	v1alpha1_core "kmesh.net/kmesh/pkg/kube/apis/kmeshnodeinfo/v1alpha1"
+	v1alpha1_clientset "kmesh.net/kmesh/pkg/kube/exnodeinfo/clientset/versioned/typed/kmeshnodeinfo/v1alpha1"
+	informer "kmesh.net/kmesh/pkg/kube/exnodeinfo/informers/externalversions"
+	v1alpha1_informers "kmesh.net/kmesh/pkg/kube/exnodeinfo/informers/externalversions/kmeshnodeinfo/v1alpha1"
+	"kmesh.net/kmesh/pkg/utils"
+)
+
+const (
+	MaxRetries   = 5
+	ActionAdd    = "add"
+	ActionDelete = "delete"
+	ActionUpdate = "modify"
+)
+
+const (
+	KmeshNodeInfoMapPath = "/sys/fs/bpf/bpf_kmesh_workload/map/map_of_nodeinfo"
+)
+
+type QueueItem struct {
+	name   string
+	action string
+}
+
+type kmeshNodeInfoMapElem struct {
+	spi    uint32
+	nodeid uint16
+	_      uint16
+}
+
+type lpm_key struct {
+	trie_key uint32
+	ip       [4]uint32
+}
+
+type IpsecController struct {
+	factory       informer.SharedInformerFactory
+	informer      v1alpha1_informers.KmeshNodeInfoInformer
+	queue         workqueue.TypedRateLimitingInterface[any]
+	kniClient     v1alpha1_clientset.KmeshNodeInfoInterface
+	kmeshNodeInfo v1alpha1_core.KmeshNodeInfo
+	ipsecHandler  *utils.IpSecHandler
+	myNode        *v1.Node
+}
+
+func NewIPsecController(k8sClientSet kubernetes.Interface) (*IpsecController, error) {
+	clientSet, err := kube.GetKmeshNodeInfoClient()
+	if err != nil {
+		err = fmt.Errorf("failed to get kmesh node info client: %v", err)
+		return nil, err
+	}
+	factroy := informer.NewSharedInformerFactory(clientSet, time.Second*0)
+
+	ipsecController := &IpsecController{
+		factory:      factroy,
+		informer:     factroy.Kmeshnodeinfo().V1alpha1().KmeshNodeInfos(),
+		queue:        workqueue.NewTypedRateLimitingQueue(workqueue.DefaultTypedControllerRateLimiter[any]()),
+		kniClient:    clientSet.KmeshnodeinfoV1alpha1().KmeshNodeInfos("kmesh-system"),
+		ipsecHandler: utils.NewIpSecHandler(),
+	}
+
+	if _, err := ipsecController.informer.Informer().AddEventHandler(cache.ResourceEventHandlerFuncs{
+		AddFunc: func(obj interface{}) {
+			ipsecController.handleKNIAddFunc(obj)
+		},
+		UpdateFunc: func(oldObj, newObj interface{}) {
+			ipsecController.handleKNIUpdateFunc(oldObj, newObj)
+		},
+		DeleteFunc: func(obj interface{}) {
+			ipsecController.handleKNIDeleteFunc(obj)
+		},
+	}); err != nil {
+		return nil, fmt.Errorf("failed to add event handler to kmeshnodeinfoInformer: %v", err)
+	}
+
+	myNodeName := os.Getenv("NODE_NAME")
+	myNode, err := k8sClientSet.CoreV1().Nodes().Get(context.TODO(), myNodeName, metav1.GetOptions{})
+	if err != nil {
+		return nil, fmt.Errorf("failed to get kmesh node info from k8s: %v", err)
+	}
+	ipsecController.myNode = myNode
+	return ipsecController, nil
+}
+
+func (ic *IpsecController) handleOtherNodeInfo(target *v1alpha1_core.KmeshNodeInfo) error {
```

Comment:
```suggestion
func (c *IpsecController) handleOtherNodeInfo(target *v1alpha1_core.KmeshNodeInfo) error {
```

change all receriver with `c`